### PR TITLE
Agp invoiceitems

### PR DESCRIPTION
**Pull request by agpiermarini**
invoice item and invoice item repository classes, plus respective methods and tests

### CODE REVIEW

File: lib/invoice_item_repository.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  def find_by_id(id)
+    @invoice_items.select { |invoice_item| invoice_item.id == id }.first
+  end
+
+  def find_all_by_item_id(item_id)
+    @invoice_items.select { |invoice_item| invoice_item.item_id == item_id }
+  end
+
+  def find_all_by_invoice_id(invoice_id)
+    @invoice_items.select { |invoice_item| invoice_item.invoice_id == invoice_id }
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:40]: Will refactor later if necessary.

[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:40]: Will refactor later if necessary.

File: lib/invoice_item_repository.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
+require 'bigdecimal'
+require 'time'
+require_relative 'invoice_item'
+require_relative 'data_analyst'
+# This is the InvoiceItemRepository class
+class InvoiceItemRepository
+  attr_reader :invoice_items, :parent
+  def initialize(csv, parent)
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:8]: Will refactor later if necessary

[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:8]: Will refactor later if necessary.

File: lib/invoice_item_repository.rb [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    @invoice_items = []
+    @parent        = parent
+    DataAnalyst.find_invoice_items(csv).each do |attribute|
+      @invoice_items << InvoiceItem.new(id:          attribute[0].to_i,
+                                        item_id:     attribute[1].to_i,
+                                        invoice_id:  attribute[2].to_i,
+                                        quantity:    attribute[3].to_i,
+                                        unit_price:  BigDecimal(attribute[4].to_i) / 100,
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:16]: Will refactor later if necessary.

[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:16]: Will refactor later if necessary.

File: lib/invoice_item_repository.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
+require 'bigdecimal'
+require 'time'
+require_relative 'invoice_item'
+require_relative 'data_analyst'
+# This is the InvoiceItemRepository class
+class InvoiceItemRepository
+  attr_reader :invoice_items, :parent
+  def initialize(csv, parent)
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/invoice_item_repository.rb:8]: Will refactor later if necessary.

**Comment by CaptainSpectacular** [comment]
Approved

**Pull request merged**